plugins/policy-recommendation/policy_recommendation_job_test.py

```diff
@@ -0,0 +1,1215 @@
+                "metadata": {"name": "recommend-k8s-np-OiBQn", "namespace": "antrea-test"},
+                "spec": {
+                    "egress": [
+                        {
+                            "ports": [{"port": 5201, "protocol": "TCP"}],
+                            "to": [
+                                {
+                                    "namespaceSelector": {
+                                        "matchLabels": {"name": "antrea-test"}
+                                    },
+                                    "podSelector": {"matchLabels": {"podname": "perftest-c"}},
+                                }
+                            ],
+                        },
+                        {
+                            "ports": [{"port": 5201, "protocol": "TCP"}],
+                            "to": [
+                                {
+                                    "namespaceSelector": {
+                                        "matchLabels": {"name": "antrea-test"}
+                                    },
+                                    "podSelector": {"matchLabels": {"podname": "perftest-b"}},
+                                }
+                            ],
+                        },
+                        {
+                            "ports": [{"port": 80, "protocol": "TCP"}],
+                            "to": [{"ipBlock": {"cidr": "192.168.0.1/32"}}],
+                        },
+                    ],
+                    "ingress": [],
+                    "podSelector": {"matchLabels": {"podname": "perftest-a"}},
+                    "policyTypes": ["Egress"],
+                },
+            },
+        }
+    )
+])
+def test_recommend_k8s_policies(spark_session, flows_input, expected_policies):
+    test_df = spark_session.createDataFrame(
+        flows_input, FLOW_TABLE_COLUMNS + ["flowType"]
+    )
+    recommend_k8s_polices_yamls = recommend_k8s_policies(test_df)
+    recommend_k8s_polices_dicts = [
+        yaml.load(i, Loader=yaml.FullLoader) for i in recommend_k8s_polices_yamls
+    ]
+    assert len(recommend_k8s_polices_dicts) == len(expected_policies)
+    for policy in recommend_k8s_polices_dicts:
+        assert (
+            "spec" in policy
+            and "podSelector" in policy["spec"]
+            and "matchLabels" in policy["spec"]["podSelector"]
+            and "podname" in policy["spec"]["podSelector"]["matchLabels"]
+        )
+        podname = policy["spec"]["podSelector"]["matchLabels"]["podname"]
+        expect_policy = expected_policies[podname]
+        assert (
+            "metadata" in policy 
+            and "name" in policy["metadata"]
+            and policy["metadata"]["name"].startswith("recommend-k8s-np-") == True
+        )
+        policy["metadata"]["name"] = expect_policy["metadata"]["name"]
+        assert policy == expect_policy
+
+
+@pytest.mark.parametrize("test_input, expected_policies", [
+    (
+        (flows_input, 1, True, False),
+        {
```

Comment:
Should we put flow data and Json objects into other input/output files? Latter, if we update the NP logic and change the input/result, maybe it would be simple to update the files directly and keep the code unchaned.

Reply:
Hi Ziyou, yes we could definitely do that, just wondering whether it's a best practice or not.
If we choose to put test input and expected output into files, there will be a lot of .txt and .json files (at least two files for each test case). Is this our convention in other Python projects?

Reply:
For test purpose, I think it is fine and it is also possiable for us to reuse the same test code for multi test cases.

Reply:
For test purpose, I think it is fine and it is also possiable for us to reuse the same test code for multi test cases.

Reply:
These are dicts; while in most cases they can be stored as JSON, there are cases when they contain obect specs which might not be serializable.
The suggestion is valid, but maybe not to be scoped in this PR.